### PR TITLE
Add integration tests for version related methods

### PR DESCRIPTION
**Pull request by jingzhang36**
In particular, the integration test pipeline_version_api_test.go is going to be included in backend-api-integration-test afterwards.

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3174)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -63,7 +63,8 @@ require (
 	github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d // indirect
 	github.com/smartystreets/goconvey v0.0.0-20181108003508-044398e4856c // indirect
 	github.com/spf13/viper v1.3.0
-	github.com/stretchr/testify v1.2.2
+	github.com/stretchr/objx v0.2.0 // indirect
+	github.com/stretchr/testify v1.3.0
```
Comment (Bobgy) [inline go.mod]: I wonder why you needed to upgrade testify?

File: backend/test/integration/pipeline_version_api_test.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	/* ---------- List pipeline versions sort by unsupported description field. Should fail. ---------- */
+	_, _, _, err = s.pipelineClient.ListPipelineVersions(&params.ListPipelineVersionsParams{
+		PageSize:        util.Int32Pointer(2),
+		SortBy:          util.StringPointer("unknownfield"),
+		ResourceKeyID:   util.StringPointer(pipelineId),
+		ResourceKeyType: util.StringPointer("PIPELINE"),
+	})
+	assert.NotNil(t, err)
+
+	/* ---------- List pipeline version s sorted by names descend order ---------- */
```
Comment (Bobgy) [inline backend/test/integration/pipeline_version_api_test.go]:
```suggestion
	/* ---------- List pipeline versions sorted by names descend order ---------- */
```

File: go.mod (hunk 1 above)
Comment (jingzhang36) [reply go.mod]: I feel this is auto changed... I didn't make this change manually.

File: backend/test/integration/pipeline_version_api_test.go (hunk 2 above)
Comment (jingzhang36) [reply backend/test/integration/pipeline_version_api_test.go]: Done.

**Pull request merged**